src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExerciseBuildConfig.java

```diff
@@ -0,0 +1,200 @@
+    private boolean checkoutSolutionRepository;
+
+    @Column(name = "checkout_path")
+    private String checkoutPath;
+
+    @Column(name = "timeout_seconds")
+    private Integer timeoutSeconds;
+
+    @Column(name = "docker_flags")
+    private String dockerFlags;
+
+    @OneToOne(mappedBy = "buildConfig")
+    @JsonIgnoreProperties("buildConfig")
+    private ProgrammingExercise programmingExercise;
+
+    @Column(name = "testwise_coverage_enabled")
+    private boolean testwiseCoverageEnabled;
+
+    @JsonProperty("sequentialTestRuns")
+    public boolean hasSequentialTestRuns() {
+        return Objects.requireNonNullElse(sequentialTestRuns, false);
+    }
+
+    public void setSequentialTestRuns(Boolean sequentialTestRuns) {
+        this.sequentialTestRuns = sequentialTestRuns;
+    }
+
+    /**
+     * Getter for the stored default branch of the exercise.
+     * Use {@link AbstractVersionControlService#getOrRetrieveBranchOfExercise(ProgrammingExercise)} if you are not sure that the value was already set in the Artemis database
+     *
+     * @return the name of the default branch or null if not yet stored in Artemis
+     */
+    public String getBranch() {
+        return branch;
+    }
+
+    public void setBranch(String branch) {
+        this.branch = branch;
+    }
+
+    /**
+     * Returns the JSON encoded custom build plan configuration
+     *
+     * @return the JSON encoded custom build plan configuration or null if the default one should be used
+     */
+    public String getBuildPlanConfiguration() {
+        return buildPlanConfiguration;
+    }
+
+    /**
+     * Sets the JSON encoded custom build plan configuration
+     *
+     * @param buildPlanConfiguration the JSON encoded custom build plan configuration
+     */
+    public void setBuildPlanConfiguration(String buildPlanConfiguration) {
+        this.buildPlanConfiguration = buildPlanConfiguration;
+    }
+
+    /**
+     * We store the bash script in the database
+     *
+     * @return the build script or null if the build script does not exist
+     */
+    public String getBuildScript() {
+        return buildScript;
+    }
+
+    /**
+     * Update the build script
+     *
+     * @param buildScript the new build script for the programming exercise
+     */
+    public void setBuildScript(String buildScript) {
+        this.buildScript = buildScript;
+    }
+
+    public Boolean getCheckoutSolutionRepository() {
+        return checkoutSolutionRepository;
+    }
+
+    public void setCheckoutSolutionRepository(Boolean checkoutSolutionRepository) {
+        this.checkoutSolutionRepository = checkoutSolutionRepository;
+    }
+
+    public String getCheckoutPath() {
+        return checkoutPath;
+    }
+
+    public void setCheckoutPath(String checkoutPath) {
+        this.checkoutPath = checkoutPath;
+    }
+
+    public Integer getTimeoutSeconds() {
+        return timeoutSeconds;
+    }
+
+    public void setTimeoutSeconds(Integer timeoutSeconds) {
```

Comment:
Again, only with a default value:
```suggestion
    public void setTimeoutSeconds(int timeoutSeconds) {
```